src/api.js

```diff
@@ -0,0 +1,797 @@
+    this.stmt = stmt1;
+    this.db = db;
+    this.pos = 1;
+    this.allocatedmem = [];
+};
+
+/* Bind values to the parameters, after having reseted the statement
+
+SQL statements can have parameters, named *'?', '?NNN', ':VVV', '@VVV', '$VVV'*,
+where NNN is a number and VVV a string.
+This function binds these parameters to the given values.
+
+*Warning*: ':', '@', and '$' are included in the parameters names
+
+## Binding values to named parameters
+@example Bind values to named parameters
+    var stmt = db.prepare("UPDATE test SET a=@newval WHERE id BETWEEN $mini AND $maxi");
+    stmt.bind({$mini:10, $maxi:20, '@newval':5});
+- Create a statement that contains parameters like '$VVV', ':VVV', '@VVV'
+- Call Statement.bind with an object as parameter
+
+## Binding values to parameters
+@example Bind values to anonymous parameters
+    var stmt = db.prepare("UPDATE test SET a=? WHERE id BETWEEN ? AND ?");
+    stmt.bind([5, 10, 20]);
+ - Create a statement that contains parameters like '?', '?NNN'
+ - Call Statement.bind with an array as parameter
+
+## Value types
+Javascript type | SQLite type
+--- | ---
+number | REAL, INTEGER
+boolean | INTEGER
+string | TEXT
+Array, Uint8Array | BLOB
+null | NULL
+@see http://www.sqlite.org/datatype3.html
+
+@see http://www.sqlite.org/lang_expr.html#varparam
+@param values [Array,Object] The values to bind
+@return [Boolean] true if it worked
+@throw [String] SQLite Error
+ */
+Statement.prototype["bind"] = function(values) {
+    if (!this.stmt) {
+        throw "Statement closed";
+    }
+    this["reset"]();
+    if (Array.isArray(values)) {
+        return this.bindFromArray(values);
+    } else {
+        return this.bindFromObject(values);
+    }
+};
+
+/* Execute the statement, fetching the the next line of result,
+that can be retrieved with [Statement.get()](#get-dynamic) .
+
+@return [Boolean] true if a row of result available
+@throw [String] SQLite Error
+ */
+Statement.prototype["step"] = function() {
+    var ret;
+    if (!this.stmt) {
+        throw "Statement closed";
+    }
+    this.pos = 1;
+    switch (ret = sqlite3_step(this.stmt)) {
+    case SQLITE_ROW:
+        return true;
+    case SQLITE_DONE:
+        return false;
+    default:
+        return this.db.handleError(ret);
+    }
+};
+
+Statement.prototype.getNumber = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_double(this.stmt, pos);
+};
+
+Statement.prototype.getString = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_column_text(this.stmt, pos);
+};
+
+Statement.prototype.getBlob = function(pos) {
+    var i, k, ptr, ref, result, size;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    size = sqlite3_column_bytes(this.stmt, pos);
+    ptr = sqlite3_column_blob(this.stmt, pos);
+    result = new Uint8Array(size);
+    for (i = k = 0, ref = size; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
```

Comment:
we can simply iterate in order

Reply:
applied suggestion in commit 43191e7a8453b83d7bd73d77754d41f1cbaf8730